### ISSUE TITLE
No iOS devices available in Simulator.app

### ISSUE THREAD

**Issue by Arjan-Zuidema** [expo/expo#27316]
### Summary

When trying to run `expo run:ios --device` I get the error `No iOS devices available in Simulator.app`

I want to run on my device, not on simulator. Why is a simulator needed?

### What platform(s) does this occur on?

iOS

### SDK Version

50

### Environment

expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.2.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.10.0 - ~/.nvm/versions/node/v20.10.0/bin/node
      Yarn: 3.4.1 - ~/.nvm/versions/node/v20.10.0/bin/yarn
      npm: 10.2.3 - ~/.nvm/versions/node/v20.10.0/bin/npm
    Managers:
      CocoaPods: 1.14.3 - /Users/arjanzuidema/.rvm/gems/ruby-2.7.2/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
    IDEs:
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    Expo Workflow: bare

### Minimal reproducible example

No need for a repo. Just a clean Xcode installation (without simulators)

**Comment by alanjhughes** [comment]
Hi @Arjan-Zuidema - Is your device connected with USB? Have you enabled developer mode on the device? https://docs.expo.dev/guides/ios-developer-mode/

**Comment by Arjan-Zuidema** [comment]
> Hi @Arjan-Zuidema - Is your device connected with USB?

Yes 

> Have you enabled developer mode on the device? https://docs.expo.dev/guides/ios-developer-mode/

Yes

I just cleaned my Mac. Removed all the xcode junk. Had to reinstall a simulator to make it work again

**Comment by byCedric** [comment]
@Arjan-Zuidema, can you share the log output when running the command below, _with_ an actual device connected.

```
DEBUG=expo:start:platforms:ios* bun expo run:ios --device
``` 

This will give us the list of commands executed to resolve the devices. We can also see if we need to "swallow" or handle an error from these commands. Based on that log, we can check if and where we need to fix this issue.

**Comment by Arjan-Zuidema** [comment]
```
$ DEBUG=expo:start:platforms:ios* npx expo run:ios --device

✔ Created native directory
✔ Updated package.json | no changes
✔ Finished prebuild
✔ Installed CocoaPods
  expo:start:platforms:ios:xcrun Running: xcrun simctl list devices --json  +0ms
  expo:start:platforms:ios:xcrun Running: xcrun simctl list devices --json  +73ms
  expo:start:platforms:ios:getBestSimulator No booted simulator matching requirements (osType: iOS). +0ms
  expo:start:platforms:ios:getBestSimulator Default simulator ID: 6B05AFA9-93C6-4B7D-9527-D44568DA99E5 +9ms
  expo:start:platforms:ios:xcrun Running: xcrun simctl list devices --json  +76ms
CommandError: No iOS devices available in Simulator.app
```

**Comment by Psycarlo** [comment]
I have the same exact issue as @Arjan-Zuidema

**Comment by hqasmei** [comment]
Same here getting it right now, clearing my xcode cache works for me

**Comment by ponny** [comment]
Same issue here.  Clearing cache and restarting didn't work.  My phone is in developer mode.  It is connected in XCode's Devices and Simulators.  I've unpaired and re-paired it a bunch of times.  Still no luck.  Simulator.app still displays no devices (right click icon, Devices >).  Will keep working at it...

**Comment by mrmuke** [comment]
I have the exact same problem

**Comment by mrmuke** [comment]
Everyone, just install the "develop for ios" packages on xcode, you must've only selected mac

**Comment by ponny** [comment]
Yep, that was it.  Open XCode, start new project, select the iOS tab.  There was an error about not having it iOS stuff installed.  Clicked the 'Get' button and it installed.

It threw me a bit because it was talking about the simulator (screenshot) but it was exactly the right thing!
[image](https://github.com/user-attachments/assets/7fe29c99-a08e-475d-8ce2-7c2fb9589826)

You're awesome @mrmuke !! 🙏

**Comment by tan-ahmed** [comment]
thanks @mrmuke

**Comment by elisaado** [comment]
We are running into the same issue. iPhone is in developer mode, connected and showing up in xcode. Building and running other apps (through xcode!) works well. Both iPhones, one on iOS 18, one on iOS 17 can't be used as a run destination in expo.

Rebooting the Mac did not work either. 

```
npx expo -v
0.18.29```

```
xcodebuild -version
Xcode 16.0
Build version 16A242d
```

**Comment by mpanac** [comment]
I have the same issue but for some reason i always do get this error although i am only targeting iOS devices and i cannot build my app:

CommandError: No tvOS devices available in Simulator.app

npx expo -v
0.18.31

xcodebuild -version
Xcode 16.1
Build version 16B40
